packages/client-github/src/index.ts

```diff
@@ -38,42 +41,448 @@ export class GitHubClient extends EventEmitter {
+
+        const response = await generateObjectV2({
+            runtime: this.runtime,
+            context,
+            modelClass: ModelClass.LARGE,
+            schema: OODASchema,
+        });
+        if (!isOODAContent(response.object)) {
+            elizaLogger.error("Invalid content in response:", response.object);
+            throw new Error("Invalid content");
+        }
+
+        const content = response.object as OODAContent;
+        elizaLogger.log("OODA content:", content);
+        if (content.action === "NOTHING") {
+            elizaLogger.log("Skipping OODA cycle as action is NOTHING");
+            return;
+        }
+        // Generate IDs with timestamp to ensure uniqueness
+        const timestamp = Date.now();
+        const userIdUUID = stringToUuid(`${this.runtime.agentId}-${timestamp}`);
+        const memoryUUID = stringToUuid(`${roomId}-${this.runtime.agentId}-${timestamp}`);
+        elizaLogger.log("Generated memory UUID:", memoryUUID);
+
+        // Create memory with retry logic
+        const newMemory: Memory = {
+            id: memoryUUID,
+            userId: userIdUUID,
+            agentId: this.runtime.agentId,
+            content: {
+                text: content.action,
+                action: content.action,
+                source: "github",
+                inReplyTo: stringToUuid(`${roomId}-${this.runtime.agentId}`)
+            },
+            roomId,
+            createdAt: timestamp,
+        };
+        elizaLogger.log("New memory to be created:", newMemory);
+
+        const responseContent = await generateMessageResponse({
+            runtime: this.runtime,
+            context,
+            modelClass: ModelClass.LARGE,
+        });
+        newMemory.content.text = responseContent.text;
+        elizaLogger.log("Generated response content:", responseContent);
+
+        try {
+            await this.runtime.messageManager.createMemory(newMemory);
+            elizaLogger.debug("Memory created successfully:", {
+                memoryId: memoryUUID,
+                action: content.action,
+                userId: this.runtime.agentId,
+            });
+        } catch (error) {
+            if (error.code === "23505") {
+                // Duplicate key error
+                elizaLogger.warn("Duplicate memory, skipping:", {
+                    memoryId: memoryUUID,
+                });
+                return;
+            }
+            elizaLogger.error("Error creating memory:", error);
+            throw error; // Re-throw other errors
+        }
+
+        // const callback: HandlerCallback = async (
+        //     content: Content,
+        //     files: any[]
+        // ) => {
+        //     elizaLogger.log("Callback called with content:", content);
+        //     return [];
+        // };
+
+        // const responseMessages = await callback(responseContent);
+        // elizaLogger.log("Response messages from callback:", responseContent);
+
```

Comment:
Not sure if we need this saw it is other clients please look at other clients to confirm